modules/material/themes/material/default/selectidp-links.twig

```diff
@@ -0,0 +1,129 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>{{ '{selectidp-links:title}'|trans }}</title>
+
+  {% include 'header.twig' %}
+
+  <script>
+    function setSelectedIdp(id) {
+      const idpInput = document.createElement('input');
+
+      idpInput.type = 'hidden';
+      idpInput.name = '{{ returnIDParam|e }}';
+      idpInput.value = id;
+
+      document.querySelector('form').appendChild(idpInput);
+
+      ga('send', 'event', 'hub', 'choice', 'IdP', id);
+    }
+
+    function clickedAnyway(idpName) {
+      ga('send', 'event', 'hub', 'choice-disabled', 'IdP', idpName);
+    }
+  </script>
+</head>
+<body>
+<div class="mdl-layout mdl-layout--fixed-header fill-viewport">
+  <header class="mdl-layout__header">
+    <div class="mdl-layout__header-row">
+      <span class="mdl-layout-title scale-to-parent">
+        {% if spName ?? '' is empty %}
+          {{ '{selectidp-links:header}'|trans }}
+        {%  else %}
+          {{ '{selectidp-links:header-for-sp}'|trans({'%spName%': spName|e}) }}
+        {% endif %}
+      </span>
+
+      <div class="mdl-layout-spacer"></div>
+
+      {% if not helpCenterUrl ?? '' is empty %}
+        <nav class="mdl-navigation">
+          <a href="{{ helpCenterUrl }}" target="_blank" class="mdl-navigation__link">
```

Comment:
(Sorry, this is as far as I got on this PR today.)

Reply:
I'm going to assume for now, that it would be correct to use the `html_attr` escape. It seems to be situational, though, because I changed the IDP name to be `html_attr` escaped and it tests because it converted "IDP 3" to "IDP&amp;#x20;3"

Reply:
> (Sorry, this is as far as I got on this PR today.)

Thank you for the suggestions thus far. I look forward to seeing more.